keps/sig-node/3327-align-by-socket/README.md

```diff
@@ -0,0 +1,493 @@
+These policy options work together to ensure an optimized cpu set is allocated for workloads running on a cluster.
+The two policy options that already exist are `full-pcpus-only`(#2625) and `distribute-cpus-across-numa` (#2902).
+With this KEP, a new `CPUManager` policy option is introduced which ensures that all CPUs on a socket are considered to be aligned.
+Thus, the `CPUManager` will send a broader set of preferred hints to `TopologyManager`, enabling the increased likelihood of the best hint to be socket aligned with respect to CPU and other devices managed by `DeviceManager`.
+
+## Motivation
+
+With the evolution of CPU architectures, the number of NUMA nodes per socket has increased.
+The devices managed by `DeviceManager` may not be uniformly distributed across all NUMA nodes.
+Thus there can be scenarios where perfect alignment between devices and CPU may not be possible.
+Latency sensitive applications desire resources to be aligned at least within the same socket if NUMA alignment is not possible for optimal performance.
+By default, the `CPUManager` prefers CPU allocations which require a minimum number of NUMA nodes.
+However, if the NUMA nodes selected for allocation are spread across sockets, it results in degraded performance.
+By ensuring the selected NUMA nodes are socket aligned, predictable performance can be achieved.
+The best possible alignment of CPUs with other resources(viz. Which are managed by `DeviceManager`) is crucial to guarantee predictable performance for latency sensitive applications.
+
+### Goals
+ * Ensure  CPUs are aligned  at socket boundary rather than NUMA node boundary.
+
+### Non-Goals
+  * Guarantee optimal NUMA allocation for cpu distribution.
+
+## Proposal
+
+We propose to add a new `CPUManager` policy option called `align-by-socket` to the  `static` policy of `CPUManager`.
+With this policy option, the `CPUManager` will prefer those hints in which all CPUs are within same socket in addition to exisiting hints which require minimum NUMA nodes. With this policy option CPUs will be considered aligned at socket boundary instead of NUMA boundary during allocation. Thus if best hint consist of NUMA nodes within one socket, `CPUManager` may try to assign available CPUs from all NUMA nodes of socket.
+
+### Risks and Mitigations
+
+The risks of adding this new feature are quite low.
+It is isolated to a specific policy option within the `CPUManager`, and is protected both by the option itself, as well as the `CPUManagerPolicyAlphaOptions` feature gate (which is disabled by default).
+
+| Risk                                             | Impact | Mitigation |
+| -------------------------------------------------| -------| ---------- |
+| Bugs in the implementation lead to kubelet crash | High   | Disable the policy option and restart the kubelet. The workload will run but CPU allocations can spread across socket in cases when allocation could have been within same socket |
```

Comment:
Are there any feedback / status back to the users if the kubelet meet its CPU alignment request or not?